### PR TITLE
Fix issue #5452 by putting in checks into the _changeCallback functio…

### LINKED ISSUE #5452: "Open With > Vega" removes line breaks and indentation
**Describe the bug**

The "Open With > Vega" feature removes line breaks and indentation from the source json file (making it difficult to develop vega json files using jupyter lab).

**To Reproduce**

1. Create a file called `vega.json` with example vega source:

    ```
    curl https://vega.github.io/vega/examples/bar-chart.vg.json > vega.json
    ```

2. Right click on the file in the jupyterlab file browser and go to "Open With > Vega" .

3. Line breaks and indentation will be removed.

**Expected behavior**
Original file formatting to be retained.

**Screenshots**

vega

**Desktop (please complete the following information):**

- OS: macOS 10.12.6
- Browser: chrome 69.0
- JupyterLab: 0.34.10

### PR DESCRIPTION
**Pull request by JohnJohnsonIII**
…n to compare data with the model, calling fromJSON only if not equal.

Fixes #5452

### CODE REVIEW

File: packages/docregistry/src/mimedocument.ts [hunk 1]
```diff
@@ -165,9 +165,13 @@ export class MimeContent extends Widget {
     }
     let data = options.data[this.mimeType];
     if (typeof data === 'string') {
-      this._context.model.fromString(data);
+      if (data === this._context.model.toString()) {
```
Comment (ian-r-rose) [inline packages/docregistry/src/mimedocument.ts]: This should be a `!==` check.

**Comment by JohnJohnsonIII** [comment]
Will correct...

**Comment by ian-r-rose** [comment]
We had a conversation with @ellisonbg about this. A longer-term fix for issues like this could involve one or both of the following:

1. Allowing mimerenderers to declare themselves as read-only. In that case, a rendermime extension could not reset the data (as was being done here by adding `image/png` mime content to the bundle.
2. Allowing mimerenderers to set additional data in a more granular fashion. As it stands, they can only update the whole bundle at once, leading to issues like the one this PR fixes.

In both of these cases, more discussion is warranted, and they would likely result in breaking changes to the rendermime interfaces. So despite the fact that this PR includes a potentially expensive check, I think this is the best way forwards in the short-term.

**Comment by ian-r-rose** [comment]
Thanks @JohnJohnsonIII!

**Pull request merged**